exploration/text-vs-code.md

```diff
@@ -0,0 +1,216 @@
+
+Most messages in any localization system do not contain any expressions, statements or variants.
+These should be expressible as easily as possible.
+
+Many messages include expressions that are to be interpolated during formatting.
+For example, a greeting like "Hello, user!" may be formatted in many locales with the `user`
+being directly set by an input variable.
+
+Sometimes, interpolated values need explicit formatting within a message.
+For example, formatting a message like "You have eaten 3.2 apples"
+may require the input numerical value
+to be formatted with an explicit `minimumFractionDigits` option.
+
+Some messages require multiple variants.
+This is often related to plural cases, such as "You have 3 new messages",
+where the value `3` is an input and the "messages" needs to correspond with its plural category.
+
+Rarely, messages needs to include leading or trailing whitespace due to
+e.g. how they will be concatenated with other text,
+or as a result of being segmented from some larger volume of text.
+
+---
+
+Users editing a simple message and who wish to add an `input` or `local` annotiation
+to the message do not wish to reformat the message extensively.
+
+Users who have messages that include leading or trailing whitespace
+want to ensure that this whitespace is included in the translatable
+text portion of the message. Which whitespace characters are displayed at runtime
+should not be surprising.
+
+## Requirements
+
+Common things should be easy, uncommon things should be possible.
+
+Developers and translators should be able to read and write the syntax easily in a text editor.
+
+Translators (and their tools) are not software engineers, so we want our syntax
+to be as simple, robust, and non-fussy as possible.
+Multiple levels of complex nesting should be avoided,
+along with any constructs that require an excessive
+level of precision on the part of non-technical users.
+
+As MessageFormat 2 will be at best a secondary language to all its users,
+it should conform to user expectations and require as little learning as possible.
+
+The syntax should avoid footguns,
+in particular as it's passed through various tools during formatting.
+
+ASCII-compatible syntax. While support for non-ASCII characters for variable names,
+values, literals, options, and the like are important, the syntax itself should
+be restricted to ASCII characters. This allows the message to be parsed
+visually by humans even when embedded in a syntax that requires escaping.
+
+Whitespace is forgiving. We _require_ the minimum amount of whitespace and allow
+users to format or change unimportant whitespace as much as they want.
+This avoids the need for translators or tools to be super pedantic about
+formatting.
+
+## Constraints
+
+Limiting the range of characters that need to be escaped in plain text is important.
+Following past precedent,
+this design doc will only consider encapsulation styles which
+start with `{` and end with `}`.
```

Comment:
This isn't actually true. We should remove it, especially since it's not germane to text-vs-code or the different PEWS handling options.

Reply:
```suggestion
```